File: solutions/kotlin/etl/1/src/main/kotlin/ETL.kt
object ETL {
    fun transform(source: Map<Int, Collection<Char>>): Map<Char, Int> {
       val result = mutableMapOf<Char,Int>()
       for((k,v) in source){ 
           for( ch in v)
               result.put(ch.toLowerCase(), k)}
        return result
    }
}

File: solutions/dart/etl/1/lib/etl.dart
class Etl {
   Map<String, int> transform(Map<String, List<String>> legacy){
     Map<String, int> result = {};
     legacy.forEach((score, letters) {
       int value = int.parse(score);
       for(var lt in letters){
          result.addAll({lt.toLowerCase(): value});
       }
     });
     return result;
   }
}
